src/features/ValueEditor/ValueEditor.tsx

```diff
@@ -22,10 +21,69 @@ import {
   FiPlus,
 } from 'react-icons/fi'
 import {LuEraser} from 'react-icons/lu'
-import {Validation} from '../../types'
+import {ZodSchema, z} from 'zod'
+import {generateErrorMessage} from 'zod-error'
 import {CopyButton} from '../CopyButton'
 import classes from './ValueEditor.module.css'
 
+const evaluateExpression = (expression: string) => {
+  try {
+    const evaluatedExpression = new Function(`return ${expression}`)()
+    return {success: true, data: evaluatedExpression} as const
+  } catch (e) {
+    return {
+      success: false,
+      error:
+        e instanceof SyntaxError
+          ? 'Invalid syntax'
+          : e instanceof ReferenceError
+            ? 'Invalid reference'
+            : 'Invalid value',
+    } as const
+  }
+}
+
+type ValidationResult =
+  | {
+      success: false
+      error: string
+    }
+  | {
+      success: true
+      parsedData: any
+    }
+
+const validateValue = (
+  schema: ZodSchema<any, z.ZodTypeDef, any> | undefined,
+  value: string,
+): ValidationResult => {
+  if (!schema) {
+    return {
+      success: false,
+      error: 'Invalid Schema',
+    }
+  }
+
+  const evaluatedValue = evaluateExpression(value)
+  if (!evaluatedValue.success) return evaluatedValue
```

Comment:
Move the return to the next line. More visible, highlights the fact that eventually you exit the function immediately.

```
  if (!evaluatedValue.success)
    return evaluatedValue
```

Reply:
this needs to be a prettier concern, not subject to code review btw

Reply:
Depends, in some context the inline syntax looks better.
Would not force either, but if you want to, newline then (in general not a code formatter fan tbh).